keyvault/shim.go

```diff
@@ -0,0 +1,158 @@
+package keyvault
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/pkcs12"
+)
+
+const (
+	pemContentType    = "application/x-pem-file"
+	pkcs12ContentType = "application/x-pkcs12"
+)
+
+type secretFetcher interface {
+	GetSecret(ctx context.Context, secretName string, opts *azsecrets.GetSecretOptions) (azsecrets.GetSecretResponse, error)
+}
+
+// Shim provides convenience methods for working with KeyVault.
+type Shim struct {
```

Comment:
Would `CertificateProvider` be closer to what this does?

Reply:
Yeah, given the api surface that is probably the most apt description. Maybe it's a bit premature to think our keyvault usage will grow beyond retrieving certs, but my original thinking was along those lines: this type is a wrapper over the az keyvault sdk that we can extend to simplify keyvault operations as needed.